### PR TITLE
switch style update and initial refactor

### PR DESCRIPTION
**Pull request by javipisobeto**

### CODE REVIEW

File: packages/react-atlas-core/src/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Checkbox extends React.PureComponent {
 
       this.setState({ checked: !this.state.checked }, function() {
         this._validationHandler(this.props.errorCallback);
-
+        debugger;
```
Comment (ericsewalson) [inline packages/react-atlas-core/src/Checkbox/Checkbox.js]: please remove debug code.

**Comment by Magneticmagnum** [comment]
Component needs to be able to handle prop changes for checked:
image
 
Styleguidst code example:

```
initialState = {checked: false}
handleToggle = () => {
    setState({ checked: !state.checked })
};

<div> 
    <Switch checked={state.checked}/>
    <Button primary onClick={handleToggle}>Toggle State</Button>
    State of active: {state.checked.toString()}
</div> 
```

**Comment by Magneticmagnum** [comment]
Inline styles are a little funny

image

**Comment by Magneticmagnum** [comment]
Style prop doesn't work when set on Slider:

image


Styleguidist code example: 
```
<div>
  <Switch style={{display: "inline-block"}} onColor="#00bbaa" offColor="#000000"/>
  <Switch style={{display: "inline-block"}} onColor="green" offColor="black"/>
</div>
```

File: packages/react-atlas-core/src/Checkbox/Checkbox.js (hunk 1 above)
Comment (javipisobeto) [reply packages/react-atlas-core/src/Checkbox/Checkbox.js]: @ericsewalson weird, this is not the latest state of the code, on my local i don't have this

**Pull request closed without merging**